java/arcs/core/storage/DirectStoreMuxer.kt

```diff
@@ -37,7 +37,7 @@ class DirectStoreMuxer<Data : CrdtData, Op : CrdtOperationAtTime, T>(
   private val writeBackProvider: WriteBackProvider,
   private val devTools: DevToolsForStorage?
 ) {
-  private val storeMutex = Mutex()
+  private val stateMutex = Mutex()
```

Comment:
Short comment here about the state that this mutex guards would be nice

Reply:
@jasonwyatt Should be be using  they `guardedBy` helper here for the protected elements?